enhancements/update/dont-require-registry-during-boot-and-upgrade.md

```diff
@@ -0,0 +1,201 @@
+---
+title: dont-require-registry-during-boot-and-upgrade
+authors:
+- "@jhernand"
+reviewers:
+- "@avishayt"
+- "@danielerez"
+- "@mrunalp"
+- "@nmagnezi"
+- "@oourfali"
+approvers:
+- "@sdodson"
+- "@zaneb"
+- "@LalatenduMohanty"
+api-approvers:
+- "@sdodson"
+- "@zaneb"
+- "@deads2k"
+- "@JoelSpeed"
+creation-date: 2023-09-21
+last-updated: 2023-09-21
+tracking-link:
+- https://issues.redhat.com/browse/RFE-4482
+see-also:
+- https://issues.redhat.com/browse/OCPBUGS-13219
+- https://github.com/openshift/enhancements/pull/1481
+- https://github.com/openshift/cluster-network-operator/pull/1803
+replaces: []
+superseded-by: []
+---
+
+# Don't require registry during boot and upgrade
+
+## Summary
+
+Ensure that clusters don't require a registry server to boot or upgrade when
+all the required images have already been pulled.
+
+## Motivation
+
+Currently during reboots and upgrades clusters may need to contact the image
+registry servers, even if the images have already been pulled. This complicates
+things for clusters that are completely disconnected or that have an slow or
+unreliable connection to the image registry servers.
+
+### User Stories
+
+#### Boot without registry
+
+As the administrator of a cluster that has all the required images already
+pulled in all the nodes, I want to be able to reboot it without requiring
+access to a registry server.
+
+### Upgrade without registry
+
+As the administrator of a cluster that has all the required images already
+pulled in all the nodes, I want to be able to upgrade it without requiring
+access to a registry server.
+
+### Goals
+
+Ensure that clusters don't require a registry server to boot or upgrade when
+all the required images have already been pulled.
```

Comment:
i think there is also a goal of:

reduce the time the cluster spends in an "upgrading" state by ensuring necessary images are pulled before the upgrade is started, this will reduce the time spent rolling out new versions of workloads during the upgrade itself.  As well as removing a potential source of upgrade failure(if the images can't be pulled beforehand, we can just not start the upgrade.  if they fail to pull during an upgrade, the upgrade gets stuck)

Reply:
I am explicitly trying to keep those goals out of this EP. Reducing the chances of failed upgrades is one of the goals of #1481. Speeding up upgrades has never been our goal.

Reply:
> Reducing the chances of failed upgrades is one of the goals of https://github.com/openshift/enhancements/pull/1481

1481 alone doesn't get you there since you also need the thing that's going to trigger+wait for the image pulling/pinning.  But it sounds like that is what is coming in your next EP rather than in this EP, so that's fine.

Reply:
Correct, the orchestration of the upgrade will be in that next EP.